<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <meta charset="utf-8">
  <meta name="generator" content="pandoc" />

    <meta name="author" content="Eric Marcon" />
  
  
  <title>entropart</title>

    <script src="index_files/jquery-1.11.3/jquery.min.js"></script>
  <link href="index_files/bootstrap-3.3.2/css/bootstrap.min.css" rel="stylesheet" />
  <script src="index_files/bootstrap-3.3.2/js/bootstrap.min.js"></script>
  <script src="index_files/bootstrap-3.3.2/shim/html5shiv.min.js"></script>
  <script src="index_files/bootstrap-3.3.2/shim/respond.min.js"></script>
  <link href="index_files/highlight-8.4/tomorrow.css" rel="stylesheet" />
  <script src="index_files/highlight-8.4/highlight.pack.js"></script>
  <link href="index_files/fontawesome-4.3.0/css/font-awesome.min.css" rel="stylesheet" />
  <script src="index_files/stickykit-1.1.1/sticky-kit.min.js"></script>
  <script src="index_files/jqueryeasing-1.3/jquery.easing.min.js"></script>
  <link href="index_files/recliner-0.2.2/recliner.css" rel="stylesheet" />
  <script src="index_files/recliner-0.2.2/recliner.min.js"></script>
  <script src="index_files/recliner-0.2.2/onload.js"></script>
  <link href="index_files/packagedocs-0.0.1/pd.css" rel="stylesheet" />
  <script src="index_files/packagedocs-0.0.1/pd.js"></script>
  <script src="index_files/packagedocs-0.0.1/pd-sticky-toc.js"></script>
  
  
  
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
</head>

<body>

  
  <header class="navbar navbar-white navbar-fixed-top" role="banner" id="header">
    <div class="container">
      <div class="navbar-header">
        <button class="navbar-toggle" type="button" data-toggle="collapse" data-target=".navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
                <a href="index.html" class="navbar-brand page-scroll">
        entropart - Entropy Partitioning to Measure Diversity
        </a>
      </div>
            <nav class="collapse navbar-collapse" role="navigation">
        <ul class="nav nav-pills pull-right">
<li class="active">
<a href='index.html'>Docs</a>
</li>
<li>
<a href='rd.html'>Package Ref</a>
</li>
<li>
<a href='https://github.com/EricMarcon/entropart'>Github<i class='fa fa-github'></i></a>
</li>
        </ul>
      </nav>
          </div>
  </header>

  <!-- Begin Body -->
  <div class="container">
    <div class="row">
            <div class="col-md-3" id="sidebar-col">
        <div id="toc">
          <ul>
          <li><a href="#introduction">Introduction</a><ul>
          <li><a href="#estimating-the-diversity-of-a-community">Estimating the diversity of a community</a></li>
          <li><a href="#estimating-the-diversity-of-a-meta-community">Estimating the diversity of a meta-community</a></li>
          </ul></li>
          <li><a href="#reference-guide">Reference guide</a><ul>
          <li><a href="#package-organization">Package organization</a></li>
          <li><a href="#species-neutral-diversity">Species-neutral diversity</a></li>
          <li><a href="#phylogenetic-diversity">Phylogenetic diversity</a></li>
          <li><a href="#similarity-based-diversity">Similarity-based diversity</a></li>
          <li><a href="#advanced-tools">Advanced tools</a></li>
          </ul></li>
          <li><a href="#phylogenies">Phylogenies</a><ul>
          <li><a href="#formats-used">Formats used</a></li>
          <li><a href="#conversion-between-classes">Conversion between classes</a></li>
          <li><a href="#newick-dendrograms">Newick dendrograms</a></li>
          </ul></li>
          <li><a href="#references">References</a></li>
          </ul>
        </div>
      </div>
      <div class="col-md-9" id="content-col">
      
<div id="content-top"></div>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
<p><em>entropart</em> is a package for <em>R</em> designed to estimate diversity based on HCDT entropy or similarity-based entropy. A short introduction to its use is in the first section of this vignette (<em>Introduction</em>). The second section is the reference guide and the last one focuses on the plylogenetic tree formats used in the package.</p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The <em>entropart</em> package allows estimating biodiversity according to the framework based on HCDT entropy, the correction of its estimation-bias <span class="citation">(Grassberger 1988; Chao and Shen 2003; Chao and Jost 2015)</span> and its transformation into equivalent numbers of species <span class="citation">(Hill 1973; Jost 2006; Marcon et al. 2014)</span>.</p>
<p>Phylogenetic or functional diversity <span class="citation">(Marcon and Hérault 2015)</span> can be estimated, considering phyloentropy as the average species-neutral diversity over slices of a phylogenetic or functional tree <span class="citation">(S. Pavoine, Love, and Bonsall 2009)</span>.</p>
<p>Similarity-based diversity <span class="citation">(Leinster and Cobbold 2012)</span> can be used to estimate <span class="citation">(Marcon, Zhang, and Hérault 2014)</span> functional diversity from a similarity or dissimilarity matrix between species without requiring building a dendrogram and thus preserving the topology of species <span class="citation">(Sandrine Pavoine, Ollier, and Dufour 2005; Podani and Schmera 2007)</span>.</p>
<p>The classical diversity estimators (Shannon and Simpson entropy) can be found in many R packages. <em>vegetarian</em> <span class="citation">(Charney and Record 2009)</span> allows calculating Hill numbers and partitioning them according to Jost’s framework. Bias correction is never available except in the <em>EntropyEstimation</em> <span class="citation">(Cao and Grabchak 2014)</span> package which provides the Zhang and Grabchak’s estimators of entropy and diversity and their asymptotic variance (not included in <em>entropart</em>).</p>
<div id="estimating-the-diversity-of-a-community" class="section level2">
<h2>Estimating the diversity of a community</h2>
<div id="community-data" class="section level3">
<h3>Community data</h3>
<p>Community data is a numeric vector containing abundances of species (the number of individual of each species) or their probabilities (the proportion of individuals of each species, summing to 1).</p>
<p>Example data is provided in the dataset <code>paracou618</code>. Let’s get the abundances of tree species in the 1-ha tropical forest plot #18 from Paracou forest station in French Guiana:</p>
<pre class="r"><code>library(&quot;entropart&quot;)
data(&quot;Paracou618&quot;)
N18 &lt;- Paracou618.MC$Nsi[, &quot;P018&quot;]</code></pre>
<p>The data in <code>Paracou618.MC</code> is a <code>MetaCommunity</code>, to be discovered later. <code>N18</code> is a vector containing the abundances of 425 tree species, among them some zero values. This is the most simple and common format to provide data to estimate diversity. It can be used directly by the functions presented here, but it may be declared explicitely as an abundance vector to plot it, and possibly fit a well-known, e.g. log-normal <span class="citation">(Preston 1948)</span>, distribution of species abundance (the red curve):</p>
<pre class="r"><code>Abd18 &lt;- as.AbdVector(N18)
plot(Abd18, Distribution=&quot;lnorm&quot;)</code></pre>
<p><img src="index_files/figure-html/PlotN18-1.png" width="624" /></p>
<pre><code>$mu
[1] 0.6843775

$sigma
[1] 0.8568455</code></pre>
<p>The parameters of the fitted distribution (here: mean and standard deviation) are returned by the function.</p>
<p>Abundance vectors can also be converted to probability vectors, summing to 1:</p>
<pre class="r"><code>P18 &lt;- as.ProbaVector(N18)</code></pre>
<p>The <code>rCommunity</code> function allows drawing random communities:</p>
<pre class="r"><code>rc &lt;- rCommunity(1, size=10000, Distribution = &quot;lseries&quot;, alpha = 30)
plot(rc, Distribution=&quot;lseries&quot;)</code></pre>
<p><img src="index_files/figure-html/rCommunity-1.png" width="624" /></p>
<pre><code>$alpha
[1] 30.17299</code></pre>
<p>The Whittaker plot of a random log-series <span class="citation">(Fisher, Corbet, and Williams 1943)</span> distribution of 10000 individuals simulated with parameter <span class="math inline">\(\alpha=30\)</span> is produced.</p>
</div>
<div id="diversity-estimation" class="section level3">
<h3>Diversity estimation</h3>
<p>The classical indices of diversity are richness (the number of species), Shannon’s and Simpson’s entropies:</p>
<pre class="r"><code>Richness(P18)</code></pre>
<pre><code>None 
 149 </code></pre>
<pre class="r"><code>Shannon(P18)</code></pre>
<pre><code>    None 
4.421358 </code></pre>
<pre class="r"><code>Simpson(P18)</code></pre>
<pre><code>     None 
0.9794563 </code></pre>
<p>When applied to a probability vector (created with <code>as.ProbaVector</code> or a numeric vector summing to 1), no estimation-bias correction is applied: this means that indices are just calculated by applying their definition function to the probabilities (that is the plugin estimator).</p>
<p>When abundances are available (a numeric vector of integer values or an object created by <code>as.ProbaVector</code>), several estimators are available <span class="citation">(Marcon 2015)</span> to address unobserved species and the non-linearity of the indices:</p>
<pre class="r"><code>Richness(Abd18)</code></pre>
<pre><code>   Chao1 
254.0888 </code></pre>
<pre class="r"><code>Shannon(Abd18)</code></pre>
<pre><code>ChaoWangJost 
     4.70651 </code></pre>
<pre class="r"><code>Simpson(Abd18)</code></pre>
<pre><code>    Lande 
0.9814969 </code></pre>
<p>The best available estimator is chosen by default: its name is returned.</p>
<p>Those indices are special cases of the Tsallis entropy <span class="citation">(1988)</span> or order <span class="math inline">\(q\)</span> (respectively <span class="math inline">\(q=0,1,2\)</span> for richness, Shannon, Simpson):</p>
<pre class="r"><code>Tsallis(Abd18,q=1)</code></pre>
<pre><code>ChaoWangJost 
     4.70651 </code></pre>
<p>Entropy should be converted to its effective number of species, i.e. the number of species with equal probabilities that would yield the observed entropy, called <span class="citation">Hill (1973)</span> numbers or simply diversity <span class="citation">(Jost 2006)</span>.</p>
<pre class="r"><code>Diversity(Abd18,q=1)</code></pre>
<pre><code>ChaoWangJost 
    110.6652 </code></pre>
<p>Diversity is the deformed exponential of order <span class="math inline">\(q\)</span> of entropy, and entropy is the deformed logarithm of of order <span class="math inline">\(q\)</span> of diversity:</p>
<pre class="r"><code>(d2 &lt;- Diversity(Abd18,q=2))</code></pre>
<pre><code>ChaoWangJost 
    54.04494 </code></pre>
<pre class="r"><code>lnq(d2, q=2)</code></pre>
<pre><code>ChaoWangJost 
   0.9814969 </code></pre>
<pre class="r"><code>(e2 &lt;-Tsallis(Abd18,q=2))</code></pre>
<pre><code>ChaoWangJost 
   0.9814969 </code></pre>
<pre class="r"><code>expq(e2, q=2)</code></pre>
<pre><code>ChaoWangJost 
    54.04494 </code></pre>
<p>Diversity can be plotted against its order to provide a diversity profile. Order 0 corresponds to richness, 1 to Shannon’s and 2 to Simpson’s diversities:</p>
<pre class="r"><code>DP &lt;- CommunityProfile(Diversity, Abd18)
plot(DP)</code></pre>
<p><img src="index_files/figure-html/DiversityProfile-1.png" width="624" /></p>
<p>If an ultrametric dendrogram describing species’ phylogeny (here, a mere taxonomy with family, genus and species) is available, phylogenetic entropy and diversity <span class="citation">(Marcon and Hérault 2015)</span> can be calculated:</p>
<pre class="r"><code>summary(PhyloDiversity(Abd18,q=1,Tree=Paracou618.Taxonomy))</code></pre>
<pre><code>alpha or gamma phylogenetic or functional diversity of order 1 
of distribution -
 with correction: Best
Phylogenetic or functional diversity was calculated according to the tree 
Paracou618.Taxonomy 

Diversity is normalized

Diversity equals: 51.98951</code></pre>
<p>With a Euclidian distance matrix between species, similarity-based diversity <span class="citation">(Leinster and Cobbold 2012; Marcon, Zhang, and Hérault 2014)</span> is available:</p>
<pre class="r"><code># Prepare the similarity matrix
DistanceMatrix &lt;- as.matrix(Paracou618.dist)
# Similarity can be 1 minus normalized distances between species
Z &lt;- 1 - DistanceMatrix/max(DistanceMatrix)
# Calculate diversity of order 2
Dqz(Abd18, q=2, Z)</code></pre>
<pre><code>    Best 
1.477898 </code></pre>
<p>Profiles of phylogenetic diversity and similarity-based diversity are obtained the same way. <code>PhyloDiversity</code> is an object with a lot of information so an intermediate function is necessary to extract its <code>$Total</code> component:</p>
<pre class="r"><code>sbDP &lt;- CommunityProfile(Dqz, Abd18, Z=Z)
pDP &lt;- CommunityProfile(function(X, ...) PhyloDiversity(X, ...)$Total, Abd18, Tree=Paracou618.Taxonomy)
plot(pDP)</code></pre>
<p><img src="index_files/figure-html/PDiversityProfile-1.png" width="624" /></p>
</div>
</div>
<div id="estimating-the-diversity-of-a-meta-community" class="section level2">
<h2>Estimating the diversity of a meta-community</h2>
<div id="meta-community-data" class="section level3">
<h3>Meta-community data</h3>
<p>A meta-community is an object defined by the package. It is a set of communities, each of them decribed by the abundance of their species and their weight. Species probabilities in the meta-community are by definition the weighted average of their probabilities in the communities.</p>
<p>The easiest way to build a meta-community consists of preparing a dataframe whose columns are communities and lines are species, and define weights in a vector (by default, all weights are equal):</p>
<pre class="r"><code>library(&quot;entropart&quot;)
(df &lt;- data.frame(C1 = c(10, 10, 10, 10), C2 = c(0, 20, 35, 5), C3 = c(25, 15, 0, 2), row.names = c(&quot;sp1&quot;, &quot;sp2&quot;, &quot;sp3&quot;, &quot;sp4&quot;)))</code></pre>
<pre><code>    C1 C2 C3
sp1 10  0 25
sp2 10 20 15
sp3 10 35  0
sp4 10  5  2</code></pre>
<pre class="r"><code>w &lt;- c(1, 2, 1)</code></pre>
<p>The <code>MetaCommunity</code> function creates the meta-community. It can be plotted:</p>
<pre class="r"><code>MC &lt;- MetaCommunity(Abundances = df, Weights = w)
plot(MC)</code></pre>
<p><img src="index_files/figure-html/MetaCommunityMC-1.png" width="624" /></p>
<p>Each shade of grey represents a species. Heights correspond to the probability of species and the width of each community is its weight.</p>
<p><code>Paracou618.MC</code> is an example meta-community provided by the package. It is made of two 1-ha communities (plots #6 and #18) of tropical forest.</p>
</div>
<div id="diversity-estimation-1" class="section level3">
<h3>Diversity estimation</h3>
<p>High level functions allow computing diversity of all communities (<span class="math inline">\(\alpha\)</span> diversity), of the meta-community (<span class="math inline">\(\gamma\)</span> diversity), and <span class="math inline">\(\beta\)</span> diversity, i.e. the number of effective communities (the number of communities with equal weights and no common species that would yield the observed <span class="math inline">\(\beta\)</span> diversity).</p>
<p>The <code>DivPart</code> function calculates everything at once, for a given order of diversity <span class="math inline">\(q\)</span>:</p>
<pre class="r"><code>p &lt;- DivPart(q = 1, MC = Paracou618.MC)
summary(p)</code></pre>
<pre><code>HCDT diversity partitioning of order 1 of metaCommunity Paracou618.MC

Alpha diversity of communities: 
    P006     P018 
66.00455 83.20917 
Total alpha diversity of the communities: 
[1] 72.88247
Beta diversity of the communities: 
    None 
1.563888 
Gamma diversity of the metacommunity: 
  None 
113.98 </code></pre>
<p>The <span class="math inline">\(\alpha\)</span> diversity of communities is 73 effective species. <span class="math inline">\(\gamma\)</span> diversity of the meta-community is 114 effective species. <span class="math inline">\(\beta\)</span> diversity is 1.56 effective communities, i.e. the two actual communities are as different from each other as 1.56 ones with equal weights and no species in common.</p>
<p>The <code>DivEst</code> function decomposes diversity and estimates confidence interval of <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity following <span class="citation">Marcon et al. (2012)</span>. If the observed species frequencies of a community are assumed to be a realization of a multinomial distribution, they can be drawn again to obtain a distribution of entropy.</p>
<pre class="r"><code>de &lt;- DivEst(q = 1, Paracou618.MC, Simulations = 100)</code></pre>
<pre><code>===========================================================================</code></pre>
<pre class="r"><code>plot(de)</code></pre>
<p><img src="index_files/figure-html/DivEst-1.png" width="624" /></p>
<p>The result is a <code>Divest</code> object which can be summarized and plotted.</p>
<p><code>DivProfile</code> calculates diversity profiles. The result is a <code>DivProfile</code> object which can be summarized and plotted.</p>
<pre class="r"><code>dp &lt;- DivProfile(, Paracou618.MC)
plot(dp)</code></pre>
<p><img src="index_files/figure-html/DivProfile-1.png" width="624" /></p>
<p>Plot #18 can be considered more diverse than plot #6 because their profiles (top right figure, plot #18 is the dotted red line, plot #6, the solid black one) do not cross <span class="citation">(Tothmeresz 1995)</span>: its diversity is systematically higher. The shape of the <span class="math inline">\(\beta\)</span> diversity profile shows that the communities are more diverse when their dominant species are considered.</p>
<p>The bootstrap confidence intervals of the values of diversity <span class="citation">(Marcon et al. 2012; Marcon et al. 2014)</span> are calculated if <code>NumberOfSimulations</code> is not 0.</p>
<p><code>DivPart</code>, <code>DivEst</code> and <code>DivProfile</code> use plugin estimators by default. To force them to apply the same estimators as community functions, the argument <code>Biased = FALSE</code> must be entered. They compute Tsallis entropy and Hill numbers by default. A dendrogram in the argument <code>Tree</code> or a similarity matrix in the argument <code>Z</code> will make them calculate phylogenetic diversity or similarity-based diversity.</p>
</div>
</div>
</div>
<div id="reference-guide" class="section level1">
<h1>Reference guide</h1>
<p>Diversity measurement can be done through a quite rigorous framework based on entropy, i.e. the amount of uncertainty calculated from the frequency distribution of a community <span class="citation">(Patil and Taillie 1982; Jost 2006; Marcon et al. 2014)</span>. Tsallis entropy, also known as HCDT entropy <span class="citation">(Havrda and Charvát 1967; Daróczy 1970; Tsallis 1988)</span>, is of particular interest <span class="citation">(Jost 2006; Marcon et al. 2014)</span> namely because it gathers the number of species, Shannon <span class="citation">(1948)</span> and Simpson <span class="citation">(1949)</span> indices of diversity into a single framework. Interpretation of entropy is not straightforward but one can easily transform it into Hill numbers <span class="citation">(Hill 1973)</span> which have many desirable properties <span class="citation">(Jost 2007)</span>: mainly, they are the number of equally-frequent species that would give the same level of diversity as the data.</p>
<p><span class="citation">Marcon and Hérault (2015)</span> generalized the duality of entropy and diversity, deriving the relation between phylogenetic or functional diversity <span class="citation">(Chao, Chiu, and Jost 2010)</span> and phylogenetic or functional entropy (we will write <em>phylodiversity</em> and <em>phyloentropy</em> for short), as introduced by <span class="citation">S. Pavoine, Love, and Bonsall (2009)</span>. Special cases are the well-known PD <span class="citation">(Faith 1992)</span> and FD <span class="citation">(Petchey and Gaston 2002)</span> indices and Rao’s <span class="citation">(1982)</span> quadratic entropy. The same relation holds between Ricotta and Szeidl’s entropy of a community <span class="citation">(Ricotta and Szeidl 2006)</span> and similarity-based diversity <span class="citation">(Leinster and Cobbold 2012)</span>.</p>
<p>The <em>entropart</em> package for R <span class="citation">(R Core Team 2018)</span> enables calculation of all these measures of diversity and entropy and their partitioning.</p>
<p>Diversity partitioning means that, in a given area, the <span class="math inline">\(\gamma\)</span> diversity <span class="math inline">\(D_{\gamma}\)</span> of all individuals found may be split into within (<span class="math inline">\(\alpha\)</span> diversity, <span class="math inline">\(D_{\alpha}\)</span>) and between (<span class="math inline">\(\beta\)</span> diversity, <span class="math inline">\(D_{\beta}\)</span>) local assemblages. <span class="math inline">\(\alpha\)</span> diversity reflects the diversity of individuals <em>in</em> local assemblages whereas <span class="math inline">\(\beta\)</span> diversity reflects the diversity <em>of</em> the local assemblages. <span class="citation">Marcon et al. (2014)</span> derived the decomposition of Tsallis <span class="math inline">\(\gamma\)</span> entropy into its <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> components, generalized to phylodiversity <span class="citation">(Marcon and Hérault 2015)</span> and similarity-based diversity <span class="citation">(Marcon, Zhang, and Hérault 2014)</span>.</p>
<p>Estimators of diversity are biased because of unseen species and also because they are not linear functions of probabilities <span class="citation">(Marcon et al. 2014)</span>. <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\gamma\)</span> diversities are underestimated by naive estimators <span class="citation">(Chao and Shen 2003; Dauby and Hardy 2012)</span>. <span class="math inline">\(\beta\)</span> diversity is severely biased too when sampling is not sufficient <span class="citation">(Beck, Holloway, and Schwanghart 2013)</span>. Bias-corrected estimators of phylodiversity have been developed by <span class="citation">Marcon and Hérault (2015)</span>. Estimators of similarity-based diversity were derived by <span class="citation">Marcon, Zhang, and Hérault (2014)</span>. The package includes them all <span class="citation">(Marcon and Hérault 2015)</span>.</p>
<p>In summary, the framework supported by the package is as follows. First, an information function is chosen to describe the amount of surprise brought by the observation of each individual. In the simplest case of species-neutral diversity, it is just a decreasing function of probability: observing an individual of a rarer species brings more surprise. Various information functions allow evaluating species-neutral, phylogenetic or functional entropy. Surprise is averaged among all individuals of a community to obtain its entropy. Entropy is systematically transformed into diversity for interpretation. Diversity is an effective number of species, i.e. the number of equally-different and equally-frequent species that would give the same entropy as the data. The average entropy of communities of an assemblage is <span class="math inline">\(\alpha\)</span> entropy, while the entropy of the assemblage is <span class="math inline">\(\gamma\)</span> entropy. Their difference is <span class="math inline">\(\beta\)</span> entropy. After transformation, <span class="math inline">\(\beta\)</span> diversity is the ratio of <span class="math inline">\(\gamma\)</span> to <span class="math inline">\(\alpha\)</span> diversity. It is an effective number of communities, i.e. the number of equally-weighted communities with no species in common (and where species are maximally distinct between communities) necessary to obtain the same diversity as the data. Estimation-bias correction is more easily applied to entropy before transforming it into diversity.</p>
<p>This framework is somehow different from that of <span class="citation">Chao, Chiu, and Jost (2014)</span> who define <span class="math inline">\(\alpha\)</span> diversity in another way <span class="citation">(see Marcon and Hérault 2015 for a detailed comparison)</span>, such that <span class="math inline">\(\alpha\)</span> entropy is not the average surprise of an assemblage. They also propose a definition of functional diversity <span class="citation">(Chiu and Chao 2014)</span> based in the information brought by pairs of individuals that is not supported in the package.</p>
<p>The successive sections of this paper presents the package features, illustrated by worked examples based on the data included in the package.</p>
<div id="package-organization" class="section level2">
<h2>Package organization</h2>
<div id="data" class="section level3">
<h3>Data</h3>
<p>Most functions of the package calculate entropy or diversity of a community or of an assemblage of communities called a <em>meta-community</em>. Community functions accept a vector of probabilities or of abundances for species data. Each element of the vector contains the probability or the number of occurrences of a species in a given community. Meta-community functions require a particular data organization in a <code>MetaCommunity</code> object described here.</p>
<p>A <code>MetaCommunity</code> is basically a list. Its main components are <code>$Nsi</code>, a matrix containing the species abundances whose lines are species, columns are communities and <code>$Wi</code>, a vector containing community weights. Creating a <code>MetaCommunity</code> object is the purpose of the <code>MetaCommunity</code> function. Arguments are a dataframe containing the number of individuals per species (lines) in each community (columns), and a vector containing the community weights. The following example creates a <code>MetaCommunity</code> made of three communities of unequal weights with 4 species. The weighted average probabilities of occurrence of species and the total number of individuals define the meta-community as the assemblage of communities.</p>
<pre class="r"><code>df &lt;- data.frame(C1=c(10, 10, 10, 10), C2=c(0, 20, 35, 5), C3=c(25, 15, 0, 2))
row.names(df) &lt;- c(&quot;sp1&quot;, &quot;sp2&quot;, &quot;sp3&quot;, &quot;sp4&quot;)
df</code></pre>
<pre><code>    C1 C2 C3
sp1 10  0 25
sp2 10 20 15
sp3 10 35  0
sp4 10  5  2</code></pre>
<pre class="r"><code>w &lt;- c(1, 2, 1)
MC &lt;- MetaCommunity(Abundances=df, Weights=w)
plot(MC)</code></pre>
<p><img src="index_files/figure-html/MetaCommunity-1.png" width="624" /></p>
<p>Communities (named C1, C2 ad C3) are represented in the left part of the figure, the metacommunity to the right. Bar widths are proportional to community weights. Species abundances are represented vertically (4 species are present in the meta-community, only 3 of them in communities C2 and C3).</p>
<p>A meta-community is partitioned into several local communities (indexed by <span class="math inline">\(i=1, 2,\dots, I\)</span>). <span class="math inline">\(n_i\)</span> individuals are sampled in community <span class="math inline">\(i\)</span>. Let <span class="math inline">\(s=1,\ 2,\dots ,S\)</span> denote the species that compose the meta-community, <span class="math inline">\(n_{s,i}\)</span> the number of individuals of species <span class="math inline">\(s\)</span> sampled in the local community <span class="math inline">\(i\)</span>, <span class="math inline">\(n_s=\sum_i{n_{s,i}}\)</span> the total number of individuals of species <span class="math inline">\(s\)</span>, <span class="math inline">\(n=\sum_s{\sum_i{n_{s,i}}}\)</span> the total number of sampled individuals. Within each community <span class="math inline">\(i\)</span>, the probability <span class="math inline">\(p_{s,i}\)</span> for an individual to belong to species <span class="math inline">\(s\)</span> is estimated by <span class="math inline">\(\hat{p}_{s,i}=n_{s,i}/{n_i}\)</span>. The same probability for the meta-community is <span class="math inline">\(p_s\)</span>.</p>
<p>Communities have a weight <span class="math inline">\(w_i\)</span>, satisfying <span class="math inline">\(p_s=\sum_i{w_i p_{s,i}}\)</span>. The commonly-used <span class="math inline">\(w_i=n_i/n\)</span> is a possible weight, but the weighting may be arbitrary (e.g. the sampled areas). The component <code>$Ps</code> of a <code>MetaCommunity</code> object contains the probability of occurrence of each species in the meta-community, calculated this way:</p>
<pre class="r"><code>MC$Ps</code></pre>
<pre><code>      sp1       sp2       sp3       sp4 
0.2113095 0.3184524 0.3541667 0.1160714 </code></pre>
<p>The number of individuals <code>$Ns</code> of a <code>MetaCommunity</code> is theoretically unknown, since communities are just samples of it. The total number of individuals is <code>$N</code>. For simplicity, it is set to the total number of individuals of all communities. If community weights are their number of individuals, <code>$Ns</code> is just the sum of the numbers of individuals per species of communities. Else, <code>$Ns</code> may contain non-integer values, respecting the probabilities <code>$Ps</code> and summing to <code>$N</code>.</p>
<p>A <code>MetaCommunity</code> can be summarized and plotted.</p>
<p>The package contains an example dataset containing the inventory of two 1-ha tropical forest plots in Paracou, French Guiana <span class="citation">(Marcon et al. 2012)</span>:</p>
<pre class="r"><code>data(&quot;Paracou618&quot;)
summary(Paracou618.MC)</code></pre>
<pre><code>Meta-community (class &#39;MetaCommunity&#39;) made of 1124 individuals in 2 
communities and 425 species. 

Its sample coverage is 0.92266748426447 

Community weights are: 
[1] 0.5720641 0.4279359
Community sample numbers of individuals are: 
P006 P018 
 643  481 
Community sample coverages are: 
     P006      P018 
0.8943859 0.8463782 </code></pre>
<p><code>Paracou618.MC</code> is a meta-community made of two communities named <em>P006</em> and <em>P018</em>, containing 425 species (their name is <em>Family_Genus_Species</em>, abbreviated to 4 characters). The values of the abundance matrix are the number of individuals of each species in each community. Sample coverage will be explained later.</p>
<p>The dataset also contains a taxonomy and a functional tree. <code>Paracou618.Taxonomy</code> is an object of class <code>phylo</code>, defined in the package <em>ape</em> <span class="citation">(Paradis, Claude, and Strimmer 2004)</span>, namely a phylogenetic tree. This example data is only a taxonomy, containing family, genus and species levels for the sake of simplicity. <code>Paracou618.Functional</code> is an object of class <code>hclust</code> containing a functional tree based on leaf, height, stem and seed functional traits <span class="citation">(Hérault and Honnay 2007; Marcon and Hérault 2015)</span>. The package also accepts any ultrametric tree of class <code>phylog</code>, from <em>ade4</em> <span class="citation">(Dray and Dufour 2007)</span> or <code>hclust</code>. <code>Paracou618.dist</code> is the distance matrix (actually a <code>dist</code> object) used to build the functional tree.</p>
<p>Numeric vectors containing species abundances (such as the <code>$Ns</code> component of <code>MetaCommunity</code>) or probabilities (such as <code>$Ps</code>) may be converted to abundance vectors (<code>AbdVector</code>) or probability vectors (<code>ProbaVector</code>) to clarify their content. By default, the <code>as.AbdVector</code> function transforms abundance values into integer if they are not (the <code>$Ns</code> components of a <code>MetaCommunity</code> is typically not an integer vector if community weights are not proportional to their numbers of individuals):</p>
<pre class="r"><code>data(&quot;Paracou618&quot;)
PAbd &lt;- as.AbdVector(Paracou618.MC$Ns)
plot(PAbd)</code></pre>
<p><img src="index_files/figure-html/AbdVector-1.png" width="624" /></p>
<p>the Paracou species distribution is plotted as a Rank-Abundance Curve (Whittaker plot).</p>
<p>The <code>as.ProbaVector</code> function transforms abundances to probabilities if necessary:</p>
<pre class="r"><code>PProba &lt;- as.ProbaVector(Paracou618.MC$Ps)</code></pre>
<p><code>AbdVector</code> and <code>ProbaVector</code> objects both are <code>SpeciesDistribution</code> objects which can be plotted.</p>
</div>
<div id="utilities" class="section level3">
<h3>Utilities</h3>
<p>The deformed logarithm formalism <span class="citation">(Tsallis 1994)</span> is very convenient to manipulate entropies. The deformed logarithm of order <span class="math inline">\(q\)</span> is defined as:</p>
<p><span class="math display">\[\ln_q{x}=\frac{x^{1-q}-1}{1-q}\]</span></p>
<p>It converges to <span class="math inline">\(\ln\)</span> when <span class="math inline">\(q\to 1\)</span>.</p>
<pre class="r"><code>curve(log(x), 0, 1, lty=1, ylab = expression(ln[q](x)))
curve(lnq(x, 0), 0, 1, lty = 2, add = TRUE)
curve(lnq(x, 2), 0, 1, lty = 3, add = TRUE)
curve(lnq(x, 3), 0, 1, lty = 4, add = TRUE)  
legend(&quot;bottomright&quot;, legend = c(expression(ln[0](x)), &quot;ln(x)&quot;, expression(ln[2](x)), expression(ln[3](x))), lty = c(2, 1, 3, 4), inset=  0.02)</code></pre>
<p><img src="index_files/figure-html/lnqCurve-1.png" width="624" /></p>
<p>The figure shows the curves of <span class="math inline">\(\ln_q{x}\)</span> for different values of <span class="math inline">\(q\)</span> between 0 and 4 (<span class="math inline">\(\ln_1{x}=\ln{x}\)</span>).</p>
<p>The inverse function of <span class="math inline">\(\ln_q{x}\)</span> is the deformed exponential:</p>
<p><span class="math display">\[e^x_q=[1+(1-q)x]^{\frac{1}{1-q}}\]</span></p>
<p>Functions of the package are <code>lnq(x, q)</code> and <code>expq(x, q)</code>.</p>
</div>
</div>
<div id="species-neutral-diversity" class="section level2">
<h2>Species-neutral diversity</h2>
<div id="community-functions" class="section level3">
<h3>Community functions</h3>
<div id="hcdt-entropy" class="section level4">
<h4>HCDT entropy</h4>
<p>Species-neutral HCDT entropy of order <span class="math inline">\(q\)</span> of a community is defined as:</p>
<p><span class="math display">\[^q\!H=\frac{1-\sum_s{p^q_s}}{q-1}=-\sum_s{p^q_s}\ln_q{p_s}=\sum_s{p_s}\ln_q{\frac{1}{p_s}}\]</span></p>
<p><span class="math inline">\(q\)</span> is the order of diversity (e.g.: 1 for Shannon). Entropy can be calculated by the <code>Tsallis</code> function. Paracou meta-community entropy of order 1 is:</p>
<pre class="r"><code>Tsallis(PProba, q = 1)</code></pre>
<pre><code>    None 
4.736023 </code></pre>
<p>For convenience, special cases of entropy of order <span class="math inline">\(q\)</span> have a clear-name function: <code>Richess</code> for <span class="math inline">\(q=0\)</span>, <code>Shannon</code> for <span class="math inline">\(q=1\)</span>, <code>Simpson</code> for <span class="math inline">\(q=2\)</span>.</p>
<pre class="r"><code>Shannon(PProba)</code></pre>
<pre><code>    None 
4.736023 </code></pre>
<p>Entropy values have no intuitive interpretation in general, except for the number of species <span class="math inline">\(^0\!H\)</span> and Simpson entropy <span class="math inline">\(^2\!H\)</span> which is the probability for two randomly chosen individuals to belong to different species.</p>
</div>
<div id="sample-coverage" class="section level4">
<h4>Sample coverage</h4>
<p>A useful indicator of sampling quality is the sample coverage <span class="citation">(Good 1953; Chao, Lee, and Chen 1988; Zhang and Huang 2007)</span>, that is to say the probability for a species of the community to be observed in the actual sample. It equals the sum of the probability of occurrences of all observed species. Its historical estimator is <span class="citation">(Good 1953)</span>:</p>
<p><span class="math display">\[\hat{C}=1-\frac{S^1}{n}\]</span></p>
<p><span class="math inline">\(S^1\)</span> is the number of singletons (species observed once) of the sample, and <span class="math inline">\(n\)</span> is its size. The estimator has been improved by taking into account the whole distribution of species <span class="citation">(Zhang and Huang 2007)</span>. The <code>Coverage</code> function calculates it, allowing to choose the estimator:</p>
<pre class="r"><code>Coverage(PAbd)</code></pre>
<pre><code>ZhangHuang 
 0.9226675 </code></pre>
<p>The sample coverage cannot be estimated from probability data: abundances are required.</p>
<p>Its interpretation is straightforward: some species have not been sampled. Their number is unknown but their total probability of occurence can be estimated accurately. Here, it is a bit less than 8%. From another point of view, the probability for an individual of the community to belong to a sampled species is <span class="math inline">\(C\)</span>: 8% of them belong to missed species. The number of missed species may be estimated by <code>Richness</code> but this is not the point here. The sample coverage is the foundation of many estimators of entropy.</p>
</div>
<div id="bias-corrected-estimators" class="section level4">
<h4>Bias corrected estimators</h4>
<p>Correction of estimation bias is used to improve the estimation of entropy despite unobserved species and also mathematical issues <span class="citation">(Bonachela, Hinrichsen, and Muñoz 2008)</span>. Bias-corrected estimators (often relying on sample coverage) are returned by functions whose names are prefixed by <code>bc</code>, such as <code>bcTsallis</code>. They are similar to the non-corrected ones but they use abundance data and propose several bias-correction techniques to select in the <code>Correction</code> argument. A <em>Best</em> correction is calculated by default, detailed in the help file of each function.</p>
<pre class="r"><code>bcTsallis(PAbd, q = 1)</code></pre>
<pre><code>ChaoWangJost 
    4.892159 </code></pre>
<p>The best correction for Tsallis entropy follows <span class="citation">Chao and Jost (2015)</span>. It combines an unbiased estimator previously derived by <span class="citation">Zhang and Grabchak (2016)</span> and an estimate of the remaining bias.</p>
<p>All community functions such as <code>Tsallis</code> are actually generic methods that can handle several types of data the appropriate way: if the first argument of the function is a <code>ProbaVector</code> (or a numeric vector summing to 1), no bias correction is applied. If it is an <code>AbdVector</code> (or an integer vector), the bias-corrected estimator is used (e.g. <code>bcTsallis</code>). Numeric vectors summing to more than 2 are considered as abundances but most bias corrections do not allow non-integer values and return a warning.</p>
<p>The different ways to use the functions are a matter of personal preference. <code>bcTsallis</code> is equivalent to <code>Tsallis</code> with an abundance vector:</p>
<pre class="r"><code>Tsallis(PAbd, q = 1)</code></pre>
<pre><code>ChaoWangJost 
    4.892159 </code></pre>
<p>whilst <code>Tsallis</code> with a probability vector does not allow bias correction:</p>
<pre class="r"><code>Tsallis(PProba, q = 1)</code></pre>
<pre><code>    None 
4.736023 </code></pre>
<p>Bias-corrected entropy is ready to be transformed into explicit diversity.</p>
</div>
<div id="effective-numbers-of-species" class="section level4">
<h4>Effective numbers of species</h4>
<p>Entropy should be converted into <em>true diversity</em> <span class="citation">(Jost 2007)</span>, i.e. effective number of species equal to <span class="citation">Hill (1973)</span> numbers:</p>
<p><span class="math display">\[^q\!D={\left(\sum_s{p^q_s}\right)}^{\frac{1}{1-q}}\]</span></p>
<p>This can be done by the deformed exponential function, or using directly the <code>Diversity</code> or <code>bcDiversity</code> functions (equal to the deformed exponential of order <span class="math inline">\(q\)</span> of <code>Tsallis</code> or <code>bcTsallis</code>)</p>
<pre class="r"><code>expq(Tsallis(PAbd, q = 2), q = 2)</code></pre>
<pre><code>ChaoWangJost 
    73.13163 </code></pre>
<pre class="r"><code>Diversity(PAbd, q = 2)</code></pre>
<pre><code>ChaoWangJost 
    73.13163 </code></pre>
<p>The effective number of species of the Paracou dataset is estimated to be 73 after bias correction (rather than 69 without it). It means that a community made of 73 equally-frequent species has the same Simpson entropy as the actual one. This is much less than the actual 425 sampled species. Simpson’s entropy focuses on dominant species.</p>
</div>
<div id="hurlberts-diversity" class="section level4">
<h4>Hurlbert’s diversity</h4>
<p>Hurlbert’s index of diversity <span class="citation">(Hurlbert 1971)</span> of order <span class="math inline">\(k\)</span> is the expected number of species observed in a sample of size <span class="math inline">\(k\)</span>.</p>
<p><span class="math display">\[_k\!S = \sum_{s}{\left[ 1-\left( 1-p_s \right)^k \right]}\]</span></p>
<p>Greater values of <span class="math inline">\(k\)</span> give more importance to rare species.</p>
<p>An unibiased estimator of <span class="math inline">\(_k\!S\)</span> has been provided by Hurlbert, for values of <span class="math inline">\(k\)</span> up to the sample size <span class="math inline">\(n\)</span>:</p>
<p><span class="math display">\[_k\!\hat{S} = \sum_{s}{\left[ 1- \binom{n-n_s}{k} / \binom{n}{k} \right]}\]</span></p>
<p>The effective number of species <span class="math inline">\(_k\!D\)</span> can be found by solving the following equation <span class="citation">(Dauby and Hardy 2012)</span>:</p>
<p><span class="math display">\[_k\!S  = {_k\!D} \left[1-{\left(1-\frac{1}{_k\!D}\right)}^k\right]\]</span></p>
<p>Hurlbert’s index is calculated by the <code>Hurlbert</code> function. Its unbiased estimator is obtained by <code>bcHurlbert</code> (implicitly if an abundance vector is used). Its effective number of species is caclulated by <code>HurlbertD</code> or <code>bcHurlbertD</code>.</p>
<pre class="r"><code>Hurlbert(PProba, k = 2)</code></pre>
<pre><code>  Biased 
1.985449 </code></pre>
<pre class="r"><code>Hurlbert(PAbd, k = 2)</code></pre>
<pre><code>Unbiased 
1.986326 </code></pre>
<pre class="r"><code>HurlbertD(PAbd, k = 2)</code></pre>
<pre><code>Unbiased 
73.13163 </code></pre>
<p>Hurlbert’s diversity of order 2 is identical to Simpson’s diversity.</p>
</div>
</div>
<div id="meta-community-functions" class="section level3">
<h3>Meta-community functions</h3>
<p>Meta-community functions allow partitioning diversity according to Patil and Taillie’s concept of diversity of a mixture <span class="citation">(Patil and Taillie 1982)</span>, i.e. <span class="math inline">\(\alpha\)</span> entropy of a meta-community is defined as the weighted average of community entropy, following <span class="citation">Routledge (1979)</span>:</p>
<p><span class="math display">\[^q\!H_{\alpha}=\sum_i w_i \,^q_iH_{\alpha}\]</span></p>
<p><span class="math inline">\(^q_iH_{\alpha}\)</span> is the entropy of community <span class="math inline">\(i\)</span>:</p>
<p><span class="math display">\[^q_i\!H_{\alpha}=\frac{1-\sum_s{p^q_{s,i}}}{q-1} =-\sum_s{p^q_{s,i}}\ln_q{p_{s,i}} =\sum_s{p_{s,i}}\ln_q{\frac{1}{p_{s,i}}}\]</span></p>
<p>Jost’s <span class="citation">(2007)</span> definition of <span class="math inline">\(\alpha\)</span> entropy is not supported explicitly in the package since it only allows partitioning of equally weighted communities. In this particular case, both definitions are identical.</p>
<p><span class="math inline">\(\gamma\)</span> entropy of the meta-community is defined as <span class="math inline">\(\alpha\)</span> entropy of a community. <span class="math inline">\(\beta\)</span> entropy, the difference between <span class="math inline">\(\gamma\)</span> and <span class="math inline">\(\alpha\)</span>, is the generalized Jensen-Shannon divergence between the species distribution of the meta-community and those of communities <span class="citation">(Marcon et al. 2014)</span>:</p>
<p><span class="math display">\[^q\!H_{\beta} =^q\!H_{\gamma}-^q\!H_{\alpha} =\sum_s{p^q_{s,i}\ln_q\frac{p_{s,i}}{p_s}} =\sum_s{p_{s,i}\ln_q\frac{p_s}{p_{s,i}}}\]</span></p>
<p><span class="math inline">\(\beta\)</span> entropy should be transformed into diversity, i.e. an effective number of communities: <span class="math display">\[^q\!D_{\beta}=e^{\frac{^q\!H_{\beta}}{1-(q-1)^q\!H_{\alpha}}}_q\]</span></p>
<div id="basic-meta-community-functions" class="section level4">
<h4>Basic meta-community functions</h4>
<p>These values can be estimated by the meta-community functions named <code>AlphaEntropy</code>, <code>AlphaDiversity</code>, <code>BetaEntropy</code>, <code>BetaDiversity</code>. They accept a <code>Metacommunity</code> and an order of diversity <span class="math inline">\(q\)</span> as arguments, and return an <code>MCentropy</code> or <code>MCdiversity</code> object which can be summarized and plotted. <code>GammaEntropy</code> and <code>GammaDiversity</code> return a number. Estimation-bias corrections are applied by default:</p>
<pre class="r"><code>e &lt;- AlphaEntropy(Paracou618.MC, q = 1)
summary(e)</code></pre>
<pre><code>Neutral alpha entropy of order 1 of metaCommunity Paracou618.MC 
with correction: Best 

Entropy of communities: 
    P006     P018 
4.384539 4.706510 
Average entropy of the communities: 
[1] 4.522322</code></pre>
<p>The Shannon <span class="math inline">\(\alpha\)</span> entropy of the meta-community is 4.52. It is the weighted average entropy of communities.</p>
<p>The estimation of the diversity of a meta-community whose numbers of individuals<code>$Ns</code> are not integer values can’t be done with most corrections, which do require integers. The Grassberger correction can be used. Community data is pooled to obtain a global inventory whose sample coverage is estimated. The Chao-Shen correction can also be applied based on this sample coverage and the actual <code>$Ps</code> values of the meta-community. Finally, the <em>Best</em> correction is the greater of the two values obtained by Chao-Shen and Grassberger <span class="citation">(Marcon et al. 2014)</span>.</p>
</div>
<div id="diversity-partition-of-a-metacommunity" class="section level4">
<h4>Diversity Partition of a metacommunity</h4>
<p>The <code>DivPart</code> function calculates everything at once. Its arguments are the same but bias correction is not applied by default. It can be, using the argument <code>Biased = FALSE</code>, and the correction chosen by the argument<code>Correction</code>. It returns a <code>DivPart</code> object which can be summarized (entropy is not printed by <code>summary</code>) and plotted:</p>
<pre class="r"><code>p &lt;- DivPart(q = 1, MC = Paracou618.MC, Biased = FALSE)
summary(p)</code></pre>
<pre><code>HCDT diversity partitioning of order 1 of metaCommunity Paracou618.MC
 with correction: ChaoWangJost
Alpha diversity of communities: 
     P006      P018 
 80.20125 110.66525 
Total alpha diversity of the communities: 
[1] 92.04908
Beta diversity of the communities: 
ChaoWangJost 
    1.447498 
Gamma diversity of the metacommunity: 
ChaoWangJost 
    133.2409 </code></pre>
<pre class="r"><code>p$CommunityAlphaEntropies</code></pre>
<pre><code>    P006     P018 
4.384539 4.706510 </code></pre>
<pre class="r"><code>plot(p)</code></pre>
<p><img src="index_files/figure-html/DivPartPlot-1.png" width="624" /></p>
<p>The <span class="math inline">\(\alpha\)</span> diversity of communities is 92 effective species (it is the exponential of the entropy calculated previously). This is more than Simpson’s diversity 73 species, calculated above) because less frequent species are taken into account. <span class="math inline">\(\gamma\)</span> diversity of the meta-community is 133 effective species. <span class="math inline">\(\beta\)</span> diversity is 1.45 effective communities, i.e. the two actual communities are as different from each other as 1.45 ones with equal weights and no species in common.</p>
<p>The figure is the plot of the diversity partition of the meta-community <code>Paracou618.MC</code>. The long rectangle of height 1 represents <span class="math inline">\(\gamma\)</span> diversity, equal to 133 effective species. The narrower and higher rectangle has the same area: its horizontal size is <span class="math inline">\(\alpha\)</span> diversity (92 effective species) and its height is <span class="math inline">\(\beta\)</span> diversity (1.45 effective communities).}</p>
</div>
<div id="diversity-estimation-of-a-metacommunity" class="section level4">
<h4>Diversity Estimation of a metacommunity</h4>
<p>The <code>DivEst</code> function decomposes diversity and estimates confidence interval of <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity following <span class="citation">Marcon et al. (2012)</span>. If the observed species frequencies of a community are assumed to be a realization of a multinomial distribution, they can be drawn again to obtain a distribution of entropy.</p>
<pre class="r"><code>de &lt;- DivEst(q = 1, Paracou618.MC, Biased = FALSE, Correction = &quot;Best&quot;, Simulations = 100)</code></pre>
<pre><code>===========================================================================</code></pre>
<pre class="r"><code>summary(de)</code></pre>
<pre><code>Diversity partitioning of order 1 of MetaCommunity Paracou618.MC
 with correction: ChaoWangJost
Alpha diversity of communities: 
     P006      P018 
 80.20125 110.66525 
Total alpha diversity of the communities: 
[1] 92.04908
Beta diversity of the communities: 
ChaoWangJost 
    1.447498 
Gamma diversity of the metacommunity: 
ChaoWangJost 
    133.2409 
Quantiles of simulations (alpha, beta and gamma diversity):
      0%       1%     2.5%       5%      10%      25%      50%      75% 
81.71923 84.29714 85.31977 86.05587 88.11714 89.98776 92.29714 94.65340 
     90%      95%    97.5%      99%     100% 
96.74340 97.27764 98.92060 99.88649 99.89239 
      0%       1%     2.5%       5%      10%      25%      50%      75% 
1.404812 1.407168 1.408263 1.410015 1.419476 1.432299 1.446128 1.461301 
     90%      95%    97.5%      99%     100% 
1.475167 1.487721 1.492045 1.502645 1.512404 
      0%       1%     2.5%       5%      10%      25%      50%      75% 
121.0360 121.9298 123.7905 125.0586 126.3524 130.2076 133.2328 136.7788 
     90%      95%    97.5%      99%     100% 
139.0588 140.3860 142.4278 142.8792 145.6888 </code></pre>
<pre class="r"><code>plot(de)</code></pre>
<p><img src="index_files/figure-html/DivEstPlot-1.png" width="624" /></p>
<p>The result is a <code>Divest</code> object which can be summarized and plotted. On the figure of the diversity estimation of the meta-community <code>Paracou618.MC</code>, <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity probability densities are plotted, with a 95% confidence interval.</p>
<p>The uncertainty of estimation is due to sampling: the distribution of the estimators corresponds to the simulated repetitions of sampling in the original multinomial distribution of species. It ignores the remaining bias of the estimator, which is unknown. Yet, except for <span class="math inline">\(q=2\)</span>, the corrected estimators <em>are</em> biased (even though much less than the non-corrected ones), especially when <span class="math inline">\(q\)</span> is small. New estimators to reduce the bias are included in the package regularly.</p>
</div>
<div id="diversity-profile-of-a-metacommunity" class="section level4">
<h4>Diversity Profile of a metacommunity</h4>
<p><code>DivProfile</code> calculates diversity profiles, i.e. the value of diversity against its order. The result is a <code>DivProfile</code> object which can be summarized and plotted.</p>
<pre class="r"><code>dp &lt;- DivProfile(seq(0, 2, 0.2), Paracou618.MC, Biased = FALSE, NumberOfSimulations = 20)</code></pre>
<pre><code>===========================================================================</code></pre>
<pre class="r"><code>summary(dp)</code></pre>
<pre><code>Diversity profile of MetaCommunity Paracou618.MC
 with correction: ChaoWangJost
Diversity against its order:
             Order Alpha Diversity Beta Diversity Gamma Diversity
ChaoWangJost   0.0       238.36351       1.321238       314.93484
ChaoWangJost   0.2       204.17972       1.323271       270.18504
ChaoWangJost   0.4       170.72117       1.335708       228.03372
ChaoWangJost   0.6       139.89024       1.361171       190.41457
ChaoWangJost   0.8       113.37058       1.399674       158.68180
ChaoWangJost   1.0        92.04908       1.447498       133.24089
ChaoWangJost   1.2        75.81819       1.498587       113.62013
ChaoWangJost   1.4        63.87372       1.546905        98.80658
ChaoWangJost   1.6        55.18979       1.588960        87.69438
ChaoWangJost   1.8        48.84485       1.623736        79.31112
ChaoWangJost   2.0        44.13559       1.651781        72.90234</code></pre>
<pre class="r"><code>plot(dp)</code></pre>
<p><img src="index_files/figure-html/DivProfilePlot-1.png" width="624" /></p>
<p>The figure shows the diversity profile of the meta-community <code>Paracou618.MC</code>. Values are the number of effective species (<span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\gamma\)</span> diversity) and the effective number of communities (<span class="math inline">\(\beta\)</span> diversity). Community P006 is represented by the solid line and community P018 by the dotted line. <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\gamma\)</span> diversity decrease from <span class="math inline">\(q=0\)</span> (number of species) to <span class="math inline">\(q=2\)</span> (Simpson diversity) by construction.}</p>
<p>Small orders of diversity give more weight to rare species. P018 can be considered more diverse than P006 because their profiles (top right of the figure) do not cross <span class="citation">(Tothmeresz 1995)</span>: its diversity is systematically higher. The shape of the <span class="math inline">\(\beta\)</span> diversity profile shows that the communities are more diverse when their dominant species are considered.</p>
<p>The bootstrap confidence intervals of the values of diversity <span class="citation">(Marcon et al. 2012; Marcon et al. 2014)</span> are calculated if <code>NumberOfSimulations</code> is not 0.</p>
</div>
<div id="alternative-functions" class="section level4">
<h4>Alternative functions</h4>
<p>Beta entropy can also be calculated by a set of functions named after the community functions, such as <code>TsallisBeta</code>, <code>bcTsallisBeta</code>, <code>SimpsonBeta</code>, etc. which require two vectors of abundances or probabilities instead of a <code>MetaCommunity</code> object: that of the community and the expected one (usually that of the meta-community). Bias correction is currently limited to Chao and Shen’s correction. The example below calculates the Shannon <span class="math inline">\(\beta\)</span> entropy of the first community of Paracou618 and the meta-community.</p>
<pre class="r"><code>ShannonBeta(Paracou618.MC$Psi[, 1], PProba)</code></pre>
<pre><code>     None 
0.3499358 </code></pre>
<p>These functions are available for particular uses, when a <code>MetaCommunity</code> is not available or not convenient to use (e.g. simulations). Meta-community functions are preferred in general.</p>
</div>
</div>
</div>
<div id="phylogenetic-diversity" class="section level2">
<h2>Phylogenetic diversity</h2>
<p>Phylogenetic or functional diversity generalizes HCDT diversity, considering the distance between species <span class="citation">(Marcon and Hérault 2015)</span>. Here, all species take place in an ultrametric phylogenetic or functional tree.</p>
<div class="figure">
<img src="Tree.png" alt="Hypothetical ultrametric tree" />
<p class="caption">Hypothetical ultrametric tree</p>
</div>
<p>The tree is cut into slices, delimited by two nodes: the hypothetical tree of the figure (a) contains three slices, delimited by two nodes. The first slice starts at the bottom of the tree and ends at the first node. In slice <span class="math inline">\(k\)</span>, <span class="math inline">\(L_k\)</span> leaves are found. The length of slices is <span class="math inline">\(T_k\)</span>. The probabilities of occurrence of the species belonging to branches that were below leaf <span class="math inline">\(l\)</span> in the original tree are summed to give the grouped probability <span class="math inline">\(u_{k,l}\)</span>. Figure (b) focuses on slice 2. The tree without slice 1 is reduced to 3 leaves. Frequencies of collapsed species are <span class="math inline">\(u_{k,l}\)</span>. Figure (c) shows slice 3 only.</p>
<p>HCDT entropy can be calculated in slice <span class="math inline">\(k\)</span>:</p>
<p><span class="math display">\[^q_k{H}=-\sum_l{u_{k,l}} \ln_q{(1/u_{k,l})}\]</span></p>
<p>Then, it is summed over the tree slices. Phyloentropy can be normalized or not. We normalize it so that it does not depend on the tree height:</p>
<p><span class="math display">\[^q\overline{H}\left( T \right) = \sum^K_{k=1}{\frac{T_k}{T}{^q_k{\!H}}}\]</span></p>
<p>Unnormalized values are multiplied by the height of the tree, such as <span class="math inline">\(^q\!\mathit{PD}(T)\)</span> <span class="citation">(Chao, Chiu, and Jost 2010)</span>.</p>
<p>Phyloentropy is calculated as HCDT entropy along the slices of the trees applying possible estimation-bias corrections, summed, possibly normalized, and finally transformed into diversity:</p>
<p><span class="math display">\[^q\overline{D}\left( T \right) = e^{^q\overline{H}\left( T \right)}_q\]</span></p>
<div id="community-functions-1" class="section level3">
<h3>Community functions</h3>
<p><code>PhyloEntropy</code> and the estimation-bias-corrected <code>bcPhyloEntropy</code> are the phylogenetic analogs of <code>Tsallis</code> and <code>bcTsallis</code>. They accept the same arguments plus an ultrametric tree of class <code>phylo</code>, <code>hclust</code> or <code>phylog</code> and <code>Normalize</code> a boolean to normalize the tree height to 1 (by default).</p>
<p>Phylogenetic diversity is calculated by <code>PhyloDiversity</code> or <code>bcPhyloDiversity</code> analogous to the species-neutral diversity functions <code>Diversity</code> and <code>bcDiversity</code>.</p>
<p>Results are either a <code>PhyloDiversity</code> or a <code>PhyloEntropy</code> object, which can be plotted and summarized.</p>
<pre class="r"><code>phd &lt;- bcPhyloDiversity(PAbd, q = 1, Tree = Paracou618.Taxonomy, Normalize = TRUE)
summary(phd)</code></pre>
<pre><code>alpha or gamma phylogenetic or functional diversity of order 1 
of distribution -
 with correction: Best
Phylogenetic or functional diversity was calculated according to the tree 
Paracou618.Taxonomy 

Diversity is normalized

Diversity equals: 54.78324</code></pre>
<pre class="r"><code>plot(phd, main = &quot;&quot;)</code></pre>
<p><img src="index_files/figure-html/PhyloDiversityPlot-1.png" width="624" /></p>
<p>The figure shows the <span class="math inline">\(\gamma\)</span> phylodiversity estimation of the meta-community <code>Paracou618.MC</code>. The effective number of taxa of Shannon diversity is plotted against the distance from the leaves of the phylogenetic tree. Here, the tree is based on a rough taxonomy, so diversity of species, genera and families are the three levels of the curve. The dotted line represents the value of phylodiversity.</p>
<p>The phylogenetic diversity of order 1 of the Paracou dataset is 55 effective species: 55 totally different species (only connected by the root of the tree) with equal probabilities would have the same entropy. It can be compared to its species-neutral diversity: 133 species. The latter is the diversity of the first slice of the tree. When going up the tree, diversity decreases because species collapse. On the figure, the diversity of the second slice, between <span class="math inline">\(T=1\)</span> and <span class="math inline">\(T=2\)</span>, is that of genera (63 effective genera) and the last slice contains20 effective families. The phylogenetic entropy of the community is the average of the entropy along slices, weighted by the slice lengths. Diversity can not be averaged the same way.</p>
<p>A less trivial phylogeny would contain many slices, resulting in as many diversity levels with respect to <span class="math inline">\(T\)</span>.</p>
<p>The <code>AllenH</code> function is close to <code>PhyloEntropy</code>: it also calculates phyloentropy but the algorithm is that of <span class="citation">Allen, Kon, and Bar-Yam (2009)</span> for <span class="math inline">\(q=1\)</span> and that of <span class="citation">Leinster and Cobbold (2012)</span> for <span class="math inline">\(q \ne 1\)</span>. It is much faster since it does not require calculating entropy for each slice of the tree but it does not allow estimation-bias correction. <code>ChaoPD</code> calculates phylodiversity according to <span class="citation">Chao, Chiu, and Jost (2010)</span>, with the same advantages and limits compared to <code>PhyloDiversity</code>.</p>
<p>For convenience, <code>PDFD</code> and <code>Rao</code> functions are provided to calculate unnormalized phyloentropy of order 0 and 2.</p>
</div>
<div id="meta-community-functions-1" class="section level3">
<h3>Meta-community functions</h3>
<p><code>DivPart</code>, <code>DivEst</code> and <code>DivProfile</code> functions return phylogenetic entropy and diversity values instead of species-neutral ones if a tree is provided in the arguments.</p>
<pre class="r"><code>dp &lt;- DivPart(q = 1, Paracou618.MC, Biased = FALSE, Correction = &quot;Best&quot;, Tree = Paracou618.Taxonomy)
summary(dp)</code></pre>
<pre><code>HCDT diversity partitioning of order 1 of metaCommunity Paracou618.MC
 with correction: ChaoWangJost ChaoWangJost ChaoWangJost
Phylogenetic or functional diversity was calculated
according to the tree 
Paracou618.Taxonomy 

Diversity is normalized 

Alpha diversity of communities: 
    P006     P018 
36.66108 51.98951 
Total alpha diversity of the communities: 
[1] 42.57235
Beta diversity of the communities: 
[1] 1.286827
Gamma diversity of the metacommunity: 
[1] 54.78324</code></pre>
<p>The decomposition is interpreted as the species-neutral one: <span class="math inline">\(\gamma\)</span> diversity is 55 effective species, made of 1.3 effective communities of 43 effective species.</p>
<p>Other meta-community functions, such as <code>AlphaEntropy</code> behave the same way:</p>
<pre class="r"><code>summary(BetaEntropy(Paracou618.MC, q = 2, Tree = Paracou618.Taxonomy, Correction = &quot;None&quot;, Normalize = FALSE))</code></pre>
<pre><code>HCDT beta entropy of order 2 of metaCommunity Paracou618.MC 
with correction: None 

Phylogenetic or functional entropy was calculated according to the tree 
Paracou618.Taxonomy 

Entropy is not normalized
Entropy of communities: 
      P006       P018 
0.04117053 0.02325883 
Average entropy of the communities: 
[1] 0.03350547</code></pre>
<p>Compare with Rao’s <code>divc</code> computed by <em>ade4</em>:</p>
<pre class="r"><code>library(&quot;ade4&quot;)
divc(as.data.frame(Paracou618.MC$Wi), disc(as.data.frame(Paracou618.MC$Nsi), Paracou618.Taxonomy$Wdist))</code></pre>
<pre><code>                   diversity
Paracou618.MC$Wi 0.009845941</code></pre>
<p>The decomposition of the diversity of meta-communities with non integer <code>$Ns</code> starts with the estimation of <span class="math inline">\(\gamma\)</span> diversity. The best estimator is found for each slice of the tree. It is then used to estimate <span class="math inline">\(\alpha\)</span> diversity.</p>
</div>
</div>
<div id="similarity-based-diversity" class="section level2">
<h2>Similarity-based diversity</h2>
<p><span class="citation">Leinster and Cobbold (2012)</span> introduced similarity-based diversity of a community <span class="math inline">\(^qD^Z\)</span>. A matrix <span class="math inline">\(\mathbf{Z}\)</span> describes the similarity between pairs of species, defined between 0 and 1. A species ordinariness is its average similarity with all species (weighted by species frequencies), including similarity with itself (equal to 1). Similarity-based diversity is the reciprocal of the generalized average of order <span class="math inline">\(q\)</span> <span class="citation">(G. Hardy, Littlewood, and Pólya 1952)</span> of the community species ordinariness.</p>
<p>The <code>Dqz</code> function calculates similarity-based diversity. Its arguments are the vector of probabilities of occurrences of the species, the order of diversity and the similarity matrix <span class="math inline">\(\mathbf{Z}\)</span>. The <code>bcDqz</code> function allows estimation-bias correction <span class="citation">(Marcon, Zhang, and Hérault 2014)</span>.</p>
<p>This example calculates the <span class="math inline">\(\gamma\)</span> diversity of the meta-community Paracou. First, the similarity matrix is calculated from the distance matrix between all pairs of species as 1 minus normalized dissimilarity.</p>
<pre class="r"><code>DistanceMatrix &lt;- as.matrix(Paracou618.dist)
Z &lt;- 1 - DistanceMatrix/max(DistanceMatrix)
bcDqz(PAbd, q = 2, Z)</code></pre>
<pre><code>   Best 
1.48295 </code></pre>
<p>If <span class="math inline">\(\mathbf{Z}\)</span> is the identity matrix, similarity-based diversity equals HCDT diversity:</p>
<pre class="r"><code>Dqz(PProba, q = 2, Z = diag(length(PProba)))</code></pre>
<pre><code>   None 
68.7215 </code></pre>
<pre class="r"><code>Diversity(PProba, q = 2)</code></pre>
<pre><code>   None 
68.7215 </code></pre>
<p>Functional diversity of order 2 is only 1.48 effective species, which is very small compared to 69 effective species for Simpson diversity. 1.48 equally-frequent species with similarity equal to 0 would have the same functional diversity as the actual community (made of 425 species). This means that species are very similar from a functional point of view. The very low values returned by <span class="math inline">\(^qD^Z\)</span> are questioned by <span class="citation">Chao, Chiu, and Jost (2014)</span> and discussed in depth by <span class="citation">Marcon, Zhang, and Hérault (2014)</span>: the choice of the similarity matrix is not trivial.</p>
<p>The similarity-based entropy of a community <span class="math inline">\(^qH^Z\)</span> [<span class="citation">Leinster and Cobbold (2012)</span>; Ricotta2006] has the same relations with diversity as HCDT entropy and Hill numbers. The <code>Hqz</code> function calculates it:</p>
<pre class="r"><code>Hqz(PProba, q = 2, Z)</code></pre>
<pre><code>     None 
0.3208152 </code></pre>
<pre class="r"><code>lnq(Dqz(PProba, q = 2, Z), q = 2)</code></pre>
<pre><code>     None 
0.3208152 </code></pre>
<p>As species-neutral entropy, <span class="math inline">\(^qH^Z\)</span> has no straightforward interpretation beyond the average surprise of a community.</p>
<p>All meta-community functions can be used to estimate similarity-based diversity: argument <code>Z</code> must be provided:</p>
<pre class="r"><code>e &lt;- AlphaEntropy(Paracou618.MC, q = 1, Z = Z)
summary(e)</code></pre>
<pre><code>Similarity-based alpha entropy of order 1 of metaCommunity Paracou618.MC 
with correction: Best 

Phylogenetic or functional entropy was calculated according to the similarity matrix 
- 

Entropy of communities: 
     P006      P018 
0.3945541 0.3934725 
Average entropy of the communities: 
[1] 0.3940912</code></pre>
<p>The <span class="math inline">\(\alpha\)</span> functional entropy of the meta-community is the average entropy of communities.</p>
</div>
<div id="advanced-tools" class="section level2">
<h2>Advanced tools</h2>
<p>The package comes with a set of tools to realize frequents tasks: run Monte-Carlo simulations on a community, quickly calculate its diversity profile, apply a function to a species distribution along a tree, and manipulate meta-communities.</p>
<div id="random-communities" class="section level3">
<h3>Random communities</h3>
<p>The <code>rCommunity</code> function allows creating random communities. Their species probability distribution can be drawn in a well-known distribution (such as a log-normal one) or obtained from the data, just by dividing abundances by the total number of individuals <span class="citation">(Marcon et al. 2012)</span>, or derived from a more sophisticated model by <span class="citation">Chao and Jost (2015)</span>. Finally, the specified number of communities are drawn in a multinomial distribution.</p>
<p>The log-normal <span class="citation">(Preston 1948)</span>, the log-series <span class="citation">(Fisher, Corbet, and Williams 1943)</span>, the geometric <span class="citation">(Motomura 1932)</span>, and the broken-stick <span class="citation">(MacArthur 1957)</span> distributions can be simulated.</p>
<p>This example code draws a single community of 1000 individuals according to a log-normal distribution with 300 species. Many species are not observed in the 1000-individual sample: the observed number of species is shown, with an estimation of the actual number (which should be 300). The simulated community is plotted (a Whittaker plot), with its log-normal distribution fitted from the data. Estimated parameters can be compared to the original ones.</p>
<pre class="r"><code>rCommunity(n = 1, size = 1000, S=300, Distribution = &quot;lnorm&quot;, sd=1) -&gt; NsRef
Richness(as.ProbaVector(NsRef))</code></pre>
<pre><code>None 
 241 </code></pre>
<pre class="r"><code>Richness(NsRef)</code></pre>
<pre><code>   Chao1 
277.2242 </code></pre>
<pre class="r"><code>plot(NsRef, Distribution=&quot;lnorm&quot;)</code></pre>
<p><img src="index_files/figure-html/rCommunityPlot-1.png" width="624" /></p>
<pre><code>$mu
[1] 1.069778

$sigma
[1] 0.8232214</code></pre>
</div>
<div id="entropy-of-monte-carlo-simulated-communities" class="section level3">
<h3>Entropy of Monte-Carlo simulated communities</h3>
<p>The <code>EntropyCI</code> function is a versatile tool to simplify simulations. Simulated communities are obtained by random draws in a multinomial distribution of species and their entropy is calculated. The arguments of <code>EntropyCI</code> are an entropy function (any entropy function of the package accepting a vector of species abundances, such as <code>bcTsallis</code>), the number of simulations to run, the observed species frequencies and the method to obtain probabilities for the multinomial distribution (the same as that of <code>rCommunity</code>).</p>
<p>The result is a numeric vector containing the entropy value of each simulated community. Entropy can be finally transformed into diversity (but it is not correct to use a diversity function in simulations because the average simulated value must be calculated and only entropy can be averaged).</p>
<p>This example shows how to use the function. First, the distribution of the <span class="math inline">\(\gamma\)</span> HCDT entropy of order 1 (Shannon entropy) of the Paracou meta-community is calculated and transformed into diversity. Then, the actual diversity is calculated and completed by the 95% confidence interval of the simulated values.</p>
<pre class="r"><code>SimulatedDiversity &lt;- expq(EntropyCI(FUN = Tsallis, Simulations = 100, Ns = PAbd, q = 1), q = 1)</code></pre>
<pre><code>===========================================================================</code></pre>
<pre class="r"><code>Diversity(PAbd, q = 1)</code></pre>
<pre><code>ChaoWangJost 
    133.2409 </code></pre>
<pre class="r"><code>quantile(SimulatedDiversity, probs = c(0.025, 0.975))</code></pre>
<pre><code>    2.5%    97.5% 
123.6827 145.9043 </code></pre>
<p>These results are identical to those of the <code>DivEst</code> function but a single community can be addressed (<code>DivEst</code> requires a <code>MetaCommunity</code>).</p>
</div>
<div id="diversity-or-entropy-profile-of-a-community" class="section level3">
<h3>Diversity or Entropy Profile of a community</h3>
<p>This function is used to calculate diversity or entropy profiles based on community functions such as <code>Tsallis</code> or <code>ChaoPD</code>. It is similar to <code>DivProfile</code> but does not require a <code>Metacommunity</code> for argument. It can compute a bootstrap confidence envelope of the estimation of the profile, like <code>EntropyCI</code>. It returns a <code>CommunityProfile</code> object which can be plotted. Profiles can be added to an existing plot by the <code>CEnvelope</code> function.</p>
<p>This example evaluates bias correction on the diversity profile of the Paracou dataset. First, diversity profiles are calculated with and without bias correction. The corrected profile is calculated with its confidence envelope:</p>
<pre class="r"><code>bcProfile &lt;- CommunityProfile(Diversity, PAbd, NumberOfSimulations = 100)</code></pre>
<pre><code>===========================================================================</code></pre>
<pre class="r"><code>Profile &lt;- CommunityProfile(Diversity, PProba)</code></pre>
<p>Then, they can be plotted altogether to obtain the <span class="math inline">\(\gamma\)</span> diversity profile of the the meta-community <code>Paracou618.MC</code>, without bias correction (dotted line) and with correction (solid line):</p>
<pre class="r"><code>plot(bcProfile)
CEnvelope(Profile, lty=3)
legend(&quot;topright&quot;, c(&quot;Bias Corrected&quot;, &quot;Biased&quot;), lty=c(1,3), inset=0.02)</code></pre>
<p><img src="index_files/figure-html/CommunityProfileFig-1.png" width="624" /></p>
</div>
<div id="applying-a-function-over-a-phylogenetic-tree" class="section level3">
<h3>Applying a Function over a Phylogenetic Tree</h3>
<p>The <code>PhyloApply</code> function is used to apply an entropy community function (generally <code>bcTsallis</code>) along a tree, the same way <code>lapply</code> works with a list.</p>
<p>This example shows how to calculate Shannon entropy along the tree containing the taxonomy to obtain species, genus and family entropy:</p>
<pre class="r"><code>pa &lt;- PhyloApply(Tree=Paracou618.Taxonomy, FUN=bcTsallis, NorP=PAbd)
summary(pa)</code></pre>
<pre><code>bcTsallis applied to - along the tree: Paracou618.Taxonomy

Results are normalized

The average value is: 4.003384

Values along the tree are:
       1        2        3 
4.892159 4.144688 2.973306 </code></pre>
<pre class="r"><code>exp(pa$Cuts)</code></pre>
<pre><code>        1         2         3 
133.24089  63.09794  19.55647 </code></pre>
<pre class="r"><code>exp(pa$Total)</code></pre>
<pre><code>[1] 54.78324</code></pre>
</div>
<div id="manipulation-of-meta-communities" class="section level3">
<h3>Manipulation of meta-communities</h3>
<p>Several meta-communities, combined in a list, can be merged two different ways. The <code>MergeMC</code> function simplifies hierarchical partitioning of diversity: it considers the aggregated data of each meta-community as a community and builds an upper-level meta-community with them. The <span class="math inline">\(\alpha\)</span> entropy of the new meta-community is the weighted average <span class="math inline">\(\gamma\)</span> entropy of the original meta-communities.</p>
<p><code>MergeC</code> combines the communities of several meta-communities to create a single meta-community containing them all. Last, <code>ShuffleMC</code> randomly shuffles communities accross meta-communities to allow simulations to test differences between meta-communities.</p>
<p>This example shows how to do this. A first meta-community is created, weights of communities are proportional to their number of individuals:</p>
<pre class="r"><code>(df &lt;- data.frame(C1 = c(10, 10, 10, 10), C2 = c(0, 20, 35, 5),
  C3 = c(25, 15, 0, 2), row.names = c(&quot;sp1&quot;, &quot;sp2&quot;, &quot;sp3&quot;, &quot;sp4&quot;)))</code></pre>
<pre><code>    C1 C2 C3
sp1 10  0 25
sp2 10 20 15
sp3 10 35  0
sp4 10  5  2</code></pre>
<pre class="r"><code>w &lt;- colSums(df)
MC1 &lt;- MetaCommunity(Abundances = df, Weights = w)</code></pre>
<p>Then a second one:</p>
<pre class="r"><code>(df &lt;- data.frame(C1 = c(10, 4), C2 = c(3, 4), row.names = c(&quot;sp1&quot;, &quot;sp5&quot;)))</code></pre>
<pre><code>    C1 C2
sp1 10  3
sp5  4  4</code></pre>
<pre class="r"><code>w &lt;- colSums(df)
MC2 &lt;- MetaCommunity(Abundances = df, Weights = w)</code></pre>
<p>They can be merged to obtain a single meta-community containing all original communities:</p>
<pre class="r"><code>mergedMC1 &lt;- MergeC(list(MC1, MC2))
mergedMC1$Nsi</code></pre>
<pre><code>    MC1.C1 MC1.C2 MC1.C3 MC2.C1 MC2.C2
sp1     10      0     25     10      3
sp2     10     20     15      0      0
sp3     10     35      0      0      0
sp4     10      5      2      0      0
sp5      0      0      0      4      4</code></pre>
<p>They can also be merged considering each of them as a community of a higher-level meta-community:</p>
<pre class="r"><code>mergedMC2 &lt;- MergeMC(list(MC1, MC2), Weights = sapply(list(MC1, MC2), function(x) (x$N)))
mergedMC2$Nsi</code></pre>
<pre><code>    MC1 MC2
sp1  35  13
sp2  45   0
sp3  45   0
sp4  17   0
sp5   0   8</code></pre>
<p>Hierarchical diversity partitioning can then be achieved:</p>
<pre class="r"><code>dpAll &lt;- DivPart(q=1, MC=mergedMC2)
summary(dpAll)</code></pre>
<pre><code>HCDT diversity partitioning of order 1 of metaCommunity -

Alpha diversity of communities: 
     MC1      MC2 
3.772161 1.943574 
Total alpha diversity of the communities: 
[1] 3.463277
Beta diversity of the communities: 
    None 
1.236351 
Gamma diversity of the metacommunity: 
    None 
4.281826 </code></pre>
<p>The <span class="math inline">\(\gamma\)</span> diversity of the top assemblage (MC1 and MC2) is 4.28 effective species, made of 1.24 effective meta-communities of 3.46 effective species. The <span class="math inline">\(\alpha\)</span> diversity of each meta-community of the top assemblage is their <span class="math inline">\(\gamma\)</span> diversity when it is partitioned in turn:</p>
<pre class="r"><code>dpMC1 &lt;- DivPart(q=1, MC=MC1)
summary(dpMC1)</code></pre>
<pre><code>HCDT diversity partitioning of order 1 of metaCommunity -

Alpha diversity of communities: 
      C1       C2       C3 
4.000000 2.429521 2.273918 
Total alpha diversity of the communities: 
[1] 2.741671
Beta diversity of the communities: 
    None 
1.375862 
Gamma diversity of the metacommunity: 
    None 
3.772161 </code></pre>
<p>The <span class="math inline">\(\gamma\)</span> diversity of MC1 is 3.77 effective species, made of 1.38 effective meta-communities of 2.74 effective species. The same decomposition can be applied to MC2.</p>
</div>
</div>
</div>
<div id="phylogenies" class="section level1">
<h1>Phylogenies</h1>
<p>Several object classes can be used in <em>entropart</em> to represent phylogenies and calculate phylogenetic diversity. They are detailed here.</p>
<div id="formats-used" class="section level2">
<h2>Formats used</h2>
<p>Supported phylogenies are classes <code>phylo</code> from package <em>ape</em>, <code>phylog</code> from package <em>ade4</em> and <code>hclust</code> from package <em>base</em>.</p>
<p>To optimize performance, a specific class <code>PPtree</code>, for “preprocessed trees”, is introduced by the package: it is basically a list containing a <code>phylo</code> and an <code>hclust</code> representation of the phylogeny and preprocessed useful statistics.</p>
<p>The arguments <code>Tree</code> or <code>PhyloTree</code> used in phylogenetic diversity estimation functions of the package may be any of those formats.</p>
</div>
<div id="conversion-between-classes" class="section level2">
<h2>Conversion between classes</h2>
<p>Let’s start from an ultrametric distance matrix. Three species are defined, with distance 1 between the first 2 and distance 2 between both and the last one.</p>
<pre class="r"><code>dSp &lt;- matrix(c(0, 1, 2, 1, 0, 2, 2, 2, 0), nrow=3, byrow=TRUE)
row.names(dSp) &lt;- colnames(dSp) &lt;- c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;)
dSp</code></pre>
<pre><code>  A B C
A 0 1 2
B 1 0 2
C 2 2 0</code></pre>
<div id="hclust" class="section level3">
<h3>hclust</h3>
<p>An <code>hclust</code> object is created by UPGMA hierarchical clustering.</p>
<pre class="r"><code>require(&quot;stats&quot;)
plot(hTree &lt;- hclust(as.dist(dSp), method=&quot;average&quot;), hang=-0.01, axes = F)
axis(2)</code></pre>
<p><img src="index_files/figure-html/hclust-1.png" width="624" /></p>
<p>Node heights are stored in <code>$height</code>.</p>
<pre class="r"><code>hTree$height</code></pre>
<pre><code>[1] 1 2</code></pre>
</div>
<div id="phylo" class="section level3">
<h3>phylo</h3>
<p>Conversion to a <code>phylo</code> object is straightforward.</p>
<pre class="r"><code>require(&quot;ape&quot;)
plot(phyloTree &lt;- as.phylo(hTree))
axis(1)</code></pre>
<p><img src="index_files/figure-html/phylo-1.png" width="624" /></p>
<p>Edge lengths have been divided by 2 during the conversion, as documented in <code>?as.phylo.hclust</code>.</p>
<pre class="r"><code>phyloTree$edge.length</code></pre>
<pre><code>[1] 1.0 0.5 0.5 0.5</code></pre>
<p>That does not suit our needs. In <em>entropart</em>, edge lengths are multiplied by 2 after conversion, so that <code>phylo</code>objects can be identical to other phylogenies.</p>
<pre class="r"><code>phyloTree$edge.length &lt;- 2*phyloTree$edge.length
plot(phyloTree)
axis(1)</code></pre>
<p><img src="index_files/figure-html/phylo3-1.png" width="624" /></p>
</div>
<div id="phylog" class="section level3">
<h3>phylog</h3>
<p>The last conversion is from <code>phylo</code> to <code>phylog</code>.</p>
<pre class="r"><code>require(&quot;ade4&quot;)
plot(phylogTree &lt;- hclust2phylog(hTree))
axis(1)</code></pre>
<p><img src="index_files/figure-html/phylog-1.png" width="624" /></p>
<p>Edge lengths are not stored directly in <code>phylog</code> objects. The <code>$droot</code> component keeps distances from tips to root. The <code>$Wdist</code> component is a <code>dist</code> (half a distance matrix without the zero-diagonal) whose values are <span class="math inline">\(d_{s,t}=\sqrt{2 \delta_{s,t}^2}\)</span>, where <span class="math inline">\(\delta_{s,t}\)</span> is the distance between species <span class="math inline">\(s\)</span> ant <span class="math inline">\(t\)</span>.</p>
<pre class="r"><code>phylogTree$droot</code></pre>
<pre><code>   C    A    B Int1 Root 
   2    2    2    1    0 </code></pre>
<pre class="r"><code>phylogTree$Wdist^2/2</code></pre>
<pre><code>  C A
A 2  
B 2 1</code></pre>
</div>
</div>
<div id="newick-dendrograms" class="section level2">
<h2>Newick dendrograms</h2>
<p>A Newick description of a dendrogram can be read into a <code>phylo</code> object.</p>
<pre class="r"><code>NewickABC &lt;- &quot;(C:2,(A:1,B:1):1);&quot;
plot(phyloABC &lt;- ape::read.tree(text=NewickABC))
axis(1)</code></pre>
<p><img src="index_files/figure-html/Newick-1.png" width="624" /></p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Allen2009">
<p>Allen, Benjamin, Mark Kon, and Yaneer Bar-Yam. 2009. “A New Phylogenetic Diversity Measure Generalizing the Shannon Index and Its Application to Phyllostomid Bats.” <em>American Naturalist</em> 174 (2): 236–43. doi:<a href="https://doi.org/10.1086/600101">10.1086/600101</a>.</p>
</div>
<div id="ref-Beck2013">
<p>Beck, Jan, Jeremy D. Holloway, and Wolfgang Schwanghart. 2013. “Undersampling and the Measurement of Beta Diversity.” <em>Methods in Ecology and Evolution</em> 4 (4): 370–82.</p>
</div>
<div id="ref-Bonachela2008">
<p>Bonachela, Juan A., Haye Hinrichsen, and Miguel A. Muñoz. 2008. “Entropy estimates of small data sets.” <em>Journal of Physics A: Mathematical and Theoretical</em> 41 (202001): 1–9. doi:<a href="https://doi.org/10.1088/1751-8113/41/20/202001">10.1088/1751-8113/41/20/202001</a>.</p>
</div>
<div id="ref-Cao2014">
<p>Cao, Lijuan, and Michael Grabchak. 2014. “EntropyEstimation.” <a href="http://cran.r-project.org/package=EntropyEstimation" class="uri">http://cran.r-project.org/package=EntropyEstimation</a>.</p>
</div>
<div id="ref-Chao2015">
<p>Chao, Anne, and Lou Jost. 2015. “Estimating Diversity and Entropy Profiles via Discovery Rates of New Species.” <em>Methods in Ecology and Evolution</em> 6 (8): 873–82.</p>
</div>
<div id="ref-Chao2003">
<p>Chao, Anne, and T. J. Shen. 2003. “Nonparametric Estimation of Shannon’s Index of Diversity When There Are Unseen Species in Sample.” <em>Environmental and Ecological Statistics</em> 10 (4): 429–43.</p>
</div>
<div id="ref-Chao2010">
<p>Chao, Anne, Chun-Huo Chiu, and Lou Jost. 2010. “Phylogenetic Diversity Measures Based on Hill Numbers.” <em>Philosophical Transactions of the Royal Society B</em> 365 (1558): 3599–3609.</p>
</div>
<div id="ref-Chao2014a">
<p>———. 2014. “Unifying Species Diversity, Phylogenetic Diversity, Functional Diversity and Related Similarity/Differentiation Measures Through Hill Numbers.” <em>Annual Review of Ecology, Evolution, and Systematics</em> 45: 297–324.</p>
</div>
<div id="ref-Chao1988">
<p>Chao, Anne, Shen-Ming Lee, and T.-C. Chen. 1988. “A Generalized Good’s Nonparametric Coverage Estimator.” <em>Chinese Journal of Mathematics</em> 16: 189–99.</p>
</div>
<div id="ref-Charney2009">
<p>Charney, Noah, and Sydne Record. 2009. “Vegetarian: Jost Diversity Measures for Community Data.” <a href="http://cran.r-project.org/package=vegetarian" class="uri">http://cran.r-project.org/package=vegetarian</a>.</p>
</div>
<div id="ref-Chiu2014b">
<p>Chiu, Chun-Huo, and Anne Chao. 2014. “Distance-Based Functional Diversity Measures and Their Decomposition: A Framework Based on Hill Numbers.” <em>PloS One</em> 9 (7): e100014.</p>
</div>
<div id="ref-Daroczy1970">
<p>Daróczy, Zoltán. 1970. “Generalized Information Functions.” <em>Information and Control</em> 16 (1): 36–51.</p>
</div>
<div id="ref-Dauby2012">
<p>Dauby, G., and O. J. Hardy. 2012. “Sampled-Based Estimation of Diversity Sensu Stricto by Transforming Hurlbert Diversities into Effective Number of Species.” <em>Ecography</em> 35 (7): 661–72.</p>
</div>
<div id="ref-Dray2007">
<p>Dray, Stéphane, and A. B. Dufour. 2007. “The Ade4 Package: Implementing the Duality Diagram for Ecologists.” <em>Journal of Statistical Software</em> 22 (4): 1–20.</p>
</div>
<div id="ref-Faith1992">
<p>Faith, D. P. 1992. “Conservation Evaluation and Phylogenetic Diversity.” <em>Biological Conservation</em> 61 (1): 1–10.</p>
</div>
<div id="ref-Fisher1943">
<p>Fisher, R. A., A. S. Corbet, and C. B. Williams. 1943. “The Relation Between the Number of Species and the Number of Individuals in a Random Sample of an Animal Population.” <em>Journal of Animal Ecology</em> 12: 42–58.</p>
</div>
<div id="ref-Good1953">
<p>Good, I. J. 1953. “On the Population Frequency of Species and the Estimation of Population Parameters.” <em>Biometrika</em> 40 (3/4): 237–64.</p>
</div>
<div id="ref-Grassberger1988">
<p>Grassberger, Peter. 1988. “Finite Sample Corrections to Entropy and Dimension Estimates.” <em>Physics Letters A</em> 128 (6–7): 369–73.</p>
</div>
<div id="ref-Hardy1952">
<p>Hardy, G.H., J.E. Littlewood, and G. Pólya. 1952. <em>Inequalities</em>. Cambridge University Press.</p>
</div>
<div id="ref-Havrda1967">
<p>Havrda, Jan, and František Charvát. 1967. “Quantification Method of Classification Processes. Concept of Structural Alpha-Entropy.” <em>Kybernetika</em> 3 (1): 30–35.</p>
</div>
<div id="ref-Herault2007">
<p>Hérault, Bruno, and Olivier Honnay. 2007. “Using Life-History Traits to Achieve a Functional Classification of Habitats.” <em>Applied Vegetation Science</em> 10 (1): 73–80.</p>
</div>
<div id="ref-Hill1973">
<p>Hill, M. O. 1973. “Diversity and Evenness: A Unifying Notation and Its Consequences.” <em>Ecology</em> 54 (2): 427–32.</p>
</div>
<div id="ref-Hurlbert1971">
<p>Hurlbert, Stuart H. 1971. “The Nonconcept of Species Diversity: A Critique and Alternative Parameters.” <em>Ecology</em> 52 (4): 577–86. doi:<a href="https://doi.org/10.2307/1934145">10.2307/1934145</a>.</p>
</div>
<div id="ref-Jost2006">
<p>Jost, Lou. 2006. “Entropy and Diversity.” <em>Oikos</em> 113 (2): 363–75.</p>
</div>
<div id="ref-Jost2007">
<p>———. 2007. “Partitioning Diversity into Independent Alpha and Beta Components.” <em>Ecology</em> 88 (10): 2427–39.</p>
</div>
<div id="ref-Leinster2012">
<p>Leinster, Tom, and Christina Cobbold. 2012. “Measuring diversity: the importance of species similarity.” <em>Ecology</em> 93 (3): 477–89. doi:<a href="https://doi.org/10.1890/10-2402.1">10.1890/10-2402.1</a>.</p>
</div>
<div id="ref-MacArthur1957">
<p>MacArthur, Robert H. 1957. “On the Relative Abundance of Bird Species.” <em>Proceedings of the National Academy of Sciences of the United States of America</em> 43 (3): 293–95. <a href="http://www.jstor.org/stable/89566" class="uri">http://www.jstor.org/stable/89566</a>.</p>
</div>
<div id="ref-Marcon2015a">
<p>Marcon, Eric. 2015. “Practical Estimation of Diversity from Abundance Data.” <em>HAL</em> 01212435 (version 2).</p>
</div>
<div id="ref-Marcon2014b">
<p>Marcon, Eric, and Bruno Hérault. 2015. “Decomposing Phylodiversity.” <em>Methods in Ecology and Evolution</em> 6 (3): 333–39. doi:<a href="https://doi.org/10.1111/2041-210X.12323">10.1111/2041-210X.12323</a>.</p>
</div>
<div id="ref-Marcon2014c">
<p>Marcon, Eric, and Bruno Hérault. 2015. “Entropart, an R Package to Partition Diversity.” <em>Journal of Statistical Software</em> 67 (8): 1–26.</p>
</div>
<div id="ref-Marcon2012a">
<p>Marcon, Eric, Bruno Hérault, Christopher Baraloto, and Gabriel Lang. 2012. “The Decomposition of Shannon’s Entropy and a Confidence Interval for Beta Diversity.” <em>Oikos</em> 121 (4): 516–22. doi:<a href="https://doi.org/10.1111/j.1600-0706.2011.19267.x">10.1111/j.1600-0706.2011.19267.x</a>.</p>
</div>
<div id="ref-Marcon2014a">
<p>Marcon, Eric, Ivan Scotti, Bruno Hérault, Vivien Rossi, and Gabriel Lang. 2014. “Generalization of the Partitioning of Shannon Diversity.” <em>PLOS One</em> 9 (3): e90289.</p>
</div>
<div id="ref-Marcon2014e">
<p>Marcon, Eric, Zhiyi Zhang, and Bruno Hérault. 2014. “The Decomposition of Similarity-Based Diversity and Its Bias Correction.” <em>HAL</em> hal-00989454 (version 1): 1–12.</p>
</div>
<div id="ref-Motomura1932">
<p>Motomura, I. 1932. “On the Statistical Treatment of Communities.” <em>Zoological Magazine</em> 44: 379–83.</p>
</div>
<div id="ref-Paradis2004">
<p>Paradis, Emmanuel, Julien Claude, and Korbinian Strimmer. 2004. “APE: Analyses of phylogenetics and evolution in R language.” <em>Bioinformatics</em> 20 (2): 289–90. doi:<a href="https://doi.org/10.1093/bioinformatics/btg412">10.1093/bioinformatics/btg412</a>.</p>
</div>
<div id="ref-Patil1982">
<p>Patil, G. P., and C. Taillie. 1982. “Diversity as a Concept and Its Measurement.” <em>Journal of the American Statistical Association</em> 77 (379): 548–61.</p>
</div>
<div id="ref-Pavoine2009">
<p>Pavoine, S., M. S. Love, and M. B. Bonsall. 2009. “Hierarchical Partitioning of Evolutionary and Ecological Patterns in the Organization of Phylogenetically-Structured Species Assemblages: Application to Rockfish (Genus: Sebastes) in the Southern California Bight.” <em>Ecology Letters</em> 12 (9): 898–908.</p>
</div>
<div id="ref-Pavoine2005a">
<p>Pavoine, Sandrine, Sébastien Ollier, and Anne-Béatrice Dufour. 2005. “Is the originality of a species measurable?” <em>Ecology Letters</em> 8: 579–86. doi:<a href="https://doi.org/10.1111/j.1461-0248.2005.00752.x">10.1111/j.1461-0248.2005.00752.x</a>.</p>
</div>
<div id="ref-Petchey2002">
<p>Petchey, Owen L., and Kevin J. Gaston. 2002. “Functional diversity (FD), species richness and community composition.” <em>Ecology Letters</em> 5: 402–11.</p>
</div>
<div id="ref-Podani2007">
<p>Podani, János, and Dénes Schmera. 2007. “How should a dendrogram-based measure of functional diversity function? A rejoinder to Petchey and Gaston.” <em>Oikos</em> 116 (8): 1427–30. doi:<a href="https://doi.org/10.1111/j.2007.0030-1299.16160.x">10.1111/j.2007.0030-1299.16160.x</a>.</p>
</div>
<div id="ref-Preston1948">
<p>Preston, F. W. 1948. “The Commonness, and Rarity, of Species.” <em>Ecology</em> 29 (3): 254–83.</p>
</div>
<div id="ref-R">
<p>R Core Team. 2018. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="http://www.r-project.org" class="uri">http://www.r-project.org</a>.</p>
</div>
<div id="ref-Rao1982">
<p>Rao, C. Radhakrishna. 1982. “Diversity and dissimilarity coefficients: a unified approach.” <em>Theoretical Population Biology</em> 21: 24–43.</p>
</div>
<div id="ref-Ricotta2006">
<p>Ricotta, Carlo, and Laszlo Szeidl. 2006. “Towards a Unifying Approach to Diversity Measures: Bridging the Gap Between the Shannon Entropy and Rao’s Quadratic Index.” <em>Theoretical Population Biology</em> 70 (3): 237–43.</p>
</div>
<div id="ref-Routledge1979">
<p>Routledge, R. D. 1979. “Diversity indices: Which ones are admissible?” <em>Journal of Theoretical Biology</em> 76 (4): 503–15. doi:<a href="https://doi.org/10.1016/0022-5193(79)90015-8">10.1016/0022-5193(79)90015-8</a>.</p>
</div>
<div id="ref-Shannon1948">
<p>Shannon, Claude E. 1948. “A Mathematical Theory of Communication.” <em>The Bell System Technical Journal</em> 27: 379–423, 623–56.</p>
</div>
<div id="ref-Simpson1949">
<p>Simpson, E. H. 1949. “Measurement of diversity.” <em>Nature</em> 163 (4148): 688.</p>
</div>
<div id="ref-Tothmeresz1995">
<p>Tothmeresz, B. 1995. “Comparison of Different Methods for Diversity Ordering.” <em>Journal of Vegetation Science</em> 6 (2): 283–90.</p>
</div>
<div id="ref-Tsallis1988">
<p>Tsallis, Constantino. 1988. “Possible Generalization of Boltzmann-Gibbs Statistics.” <em>Journal of Statistical Physics</em> 52 (1): 479–87.</p>
</div>
<div id="ref-Tsallis1994">
<p>———. 1994. “What are the numbers that experiments provide?” <em>Química Nova</em> 17 (6): 468–71. <a href="http://quimicanova.sbq.org.br/detalhe{\_}artigo.asp?id=5517" class="uri">http://quimicanova.sbq.org.br/detalhe{\_}artigo.asp?id=5517</a>.</p>
</div>
<div id="ref-Zhang2014">
<p>Zhang, Zhiyi, and Michael Grabchak. 2016. “Entropic Representation and Estimation of Diversity Indices.” <em>Journal of Nonparametric Statistics</em> 28 (3): 563–75.</p>
</div>
<div id="ref-Zhang2007">
<p>Zhang, Zhiyi, and Hongwei Huang. 2007. “Turing’s Formula Revisited.” <em>Journal of Quantitative Linguistics</em> 14 (2-3): 222–41.</p>
</div>
</div>
</div>


      </div>
    </div>
  </div>

  <div id="footer">
    <div class="container">
      <div class="col-md-6">
                <p>&copy; Eric Marcon, 2018</p>
              </div>
      <div class="col-md-6">
        <p class="pull-right">created with <a href="https://github.com/hafen/packagedocs">packagedocs</a>
                  </p>
      </div>
    </div>
  </div>

  
</body>
</html>
